library/src/test/java/com/novoda/downloadmanager/DownloadManagerTest.java

```diff
@@ -0,0 +1,281 @@
+    }
+
+    @Test
+    public void pausesBatch() {
+        downloadManager.pause(DOWNLOAD_BATCH_ID);
+
+        verify(downloadBatch).pause();
+    }
+
+    @Test
+    public void doesNotResume_whenBatchIdIsUnknown() {
+        downloadManager.pause(new LiteDownloadBatchId("unknown"));
+
+        verifyZeroInteractions(downloadBatch, additionalDownloadBatch);
+    }
+
+    @Test
+    public void doesNotResume_whenBatchIsAlreadyDownloading() {
+        given(downloadBatch.status()).willReturn(anInternalDownloadsBatchStatus().withStatus(DownloadBatchStatus.Status.DOWNLOADING).build());
+
+        downloadManager.resume(DOWNLOAD_BATCH_ID);
+
+        InOrder inOrder = inOrder(downloadBatch);
+        inOrder.verify(downloadBatch).status();
+        inOrder.verifyNoMoreInteractions();
+    }
+
+    @Test
+    public void removesBatchFromInternalList_whenResuming() {
+        given(downloadBatch.status()).willReturn(anInternalDownloadsBatchStatus().build());
+
+        downloadManager.resume(DOWNLOAD_BATCH_ID);
+
+        assertThat(downloadBatches).doesNotContainEntry(DOWNLOAD_BATCH_ID, downloadBatch);
+    }
+
+    @Test
+    public void resumesBatch() {
+        given(downloadBatch.status()).willReturn(anInternalDownloadsBatchStatus().build());
+
+        downloadManager.resume(DOWNLOAD_BATCH_ID);
+
+        verify(downloadBatch).resume();
+    }
+
+    @Test
+    public void triggersDownload_whenResumingBatch() {
+        given(downloadBatch.status()).willReturn(anInternalDownloadsBatchStatus().build());
+
+        downloadManager.resume(DOWNLOAD_BATCH_ID);
+
+        verify(downloadManagerDownloader).download(downloadBatch, downloadBatches);
+    }
+
+    @Test
+    public void doesNotDelete_whenBatchIdIsUnknown() {
+        downloadManager.delete(new LiteDownloadBatchId("unknown"));
+
+        verifyZeroInteractions(downloadBatch, additionalDownloadBatch);
+    }
+
+    @Test
+    public void removesBatchFromInternalList_whenDeleting() {
+        downloadManager.delete(DOWNLOAD_BATCH_ID);
+
+        assertThat(downloadBatches).doesNotContainEntry(DOWNLOAD_BATCH_ID, downloadBatch);
+    }
+
+    @Test
+    public void deletesBatch() {
+        downloadManager.delete(DOWNLOAD_BATCH_ID);
+
+        verify(downloadBatch).delete();
+    }
+
+    @Test
+    public void addsCallbackToInternalList() {
+        DownloadBatchCallback additionalDownloadBatchCallback = mock(DownloadBatchCallback.class);
+
+        downloadManager.addDownloadBatchCallback(additionalDownloadBatchCallback);
+
+        assertThat(downloadBatchCallbacks).contains(additionalDownloadBatchCallback);
+    }
+
+    @Test
+    public void removesCallbackFromInternalList() {
+        downloadManager.removeDownloadBatchCallback(downloadBatchCallback);
+
+        assertThat(downloadBatchCallbacks).doesNotContain(downloadBatchCallback);
+    }
+
+    @Test(timeout = 500)
+    public void waitsForServiceToExist_whenGettingAllBatchStatuses() throws InterruptedException {
```

Comment:
This one was painful. Need to argument captor the runnable to ensure that it runs on the same thread as the test, so it locks up when the internal `wait` is called. We then need to trigger a notify on another thread which will release the testing thread so that it doesn't hit the timeout and fail.